### PR TITLE
out_retry: register config map

### PR DESCRIPTION
**Pull request by aSquare14**
Signed-off-by: Atibhi Agrawal <Atibhi.Agrawal@iiitb.org>
#1672

### CODE REVIEW

**Comment by edsiper** [comment]
The logic of the config map is wrong:

- n_retry is an internal field of the context structure, not an exposed property. The property name is "retries". Note that config maps support different data types, with configmap you can get rid of the need to use atoi() or another conversion function.

File: plugins/out_retry/retry.c [hunk 1]
```diff
@@ -98,11 +90,24 @@ static int cb_retry_exit(void *data, struct flb_config *config)
     return 0;
 }
 
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+        FLB_CONFIG_MAP_STR, "retries", NULL,
```
Comment (edsiper) [inline plugins/out_retry/retry.c]: the old code specifies a default value, if you don't set one, you are breaking backward compatibility.

**Comment by edsiper** [comment]
Please test functionality,

**Comment by aSquare14** [comment]
@edsiper Could not find documentation for this plugin. How to test functionality and can I add documentation ?

**Comment by edsiper** [comment]
you can do any improvement in any area. Our documentation repository is here: github.com/fluent-bit-docs

File: plugins/out_retry/retry.c [hunk 2]
```diff
@@ -80,12 +72,12 @@ static void cb_retry_flush(const void *data, size_t bytes,
...
     ctx->count++;
 
-    if (ctx->count <= ctx->n_retry) {
-        flb_plg_debug(ctx->ins, "retry %i/%i", ctx->count, ctx->n_retry);
+    if (ctx->count <= ctx->retries) {
+        flb_debug("[retry] retry %i/%i", ctx->count, ctx->retries);
         FLB_OUTPUT_RETURN(FLB_RETRY);
     }
     else {
-        flb_plg_debug(ctx->ins, "flush", ctx->count, ctx->n_retry);
```
Comment (edsiper) [inline plugins/out_retry/retry.c]: note that this change is reverting our improvement in the logging api (same thing above)

Comment (aSquare14) [reply plugins/out_retry/retry.c]: @edsiper okay. So what changes shall I make

**Comment by aSquare14** [comment]
After enabling debug mode and running the command 
`cmake -DFLB_DEBUG=Yes -DFLB_OUT_RETRY=Yes`

Screenshot from 2020-02-25 10-55-07
Screenshot from 2020-02-25 10-55-10

**Comment by aSquare14** [comment]
@koleini Please review

**Comment by aSquare14** [comment]
@edsiper Pl review

**Comment by edsiper** [comment]
PR review:

```
$ gdb bin/fluent-bit
...
(gdb) br retry.c:74
Breakpoint 1 at 0xdbbc8: file /home/edsiper/coding/fluent-bit/plugins/out_retry/retry.c, line 75.
(gdb) run -i random -p samples=1 -o retry -p retries=10 -f 1
Starting program: /home/edsiper/coding/fluent-bit/build/bin/fluent-bit -i random -p samples=1 -o retry -p retries=10 -f 1
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Fluent Bit v1.4.0
* Copyright (C) 2019-2020 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[New Thread 0x7ffff79da700 (LWP 3277991)]
[2020/05/04 17:07:14] [ info] [storage] version=1.0.0, initializing...
[2020/05/04 17:07:14] [ info] [storage] in-memory
[2020/05/04 17:07:14] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2020/05/04 17:07:14] [ info] [engine] started (pid=3277987)
[2020/05/04 17:07:14] [ info] [sp] stream processor started

Thread 1 "fluent-bit" hit Breakpoint 1, cb_retry_flush (data=0x7ffff7199010, bytes=32, tag=0x5555559ca560 "random.0", tag_len=8, i_ins=0x5555559c32e0, out_context=0x5555559c8270, 
    config=0x5555559bc2d0) at /home/edsiper/coding/fluent-bit/plugins/out_retry/retry.c:75
75	    if (ctx->count <= ctx->retries) {
(gdb) print ctx->retries 
$1 = 0
```

- plugin don't use flb_output_config_map_set() API
- code changes breaks plugin behavior, ```retries``` is never set despite it was given a ```10``` value.

**Pull request closed without merging**